packages/core/src/utils.ts

```diff
@@ -1,16 +1,31 @@
 import * as fs from 'fs';
 import { extname } from 'path';
 import * as minimatch from 'minimatch';
-import fetch from 'node-fetch';
 import { parseYaml } from './js-yaml';
 import { env } from './env';
 import { logger, colorize } from './logger';
-import { HttpsProxyAgent } from 'https-proxy-agent';
 import * as pluralizeOne from 'pluralize';
 
 import type { HttpResolveConfig } from './config';
 import type { UserContext } from './walk';
 
+let fetch: ((input: RequestInfo | URL, init?: RequestInit) => Promise<Response>) & {
+    (input: RequestInfo | URL, init?: RequestInit): Promise<Response>;
+    (input: string | URL | globalThis.Request, init?: RequestInit): Promise<Response>;
+  },
+  ProxyAgent: (new (arg0: string) => any) | null;
+
+if (typeof window === 'undefined') {
```

Comment:
this is has to be fixed very carefully

Reply:
Its for browser support, I found some logger tests for that.
Going to test in vscode / monorepo with some verdaccio package.

Reply:
I changed it to `isBrowser` from `.env`. Or should we use some other workaround?